### ISSUE TITLE
SourceCred should report last-updated information

### ISSUE THREAD

**Issue by decentralion** [sourcecred/sourcecred#989]
When we do a cred analysis, the UI should report when the data was last sync'd. This way it's easy to see if we're looking at old data.

We will need to change the `repositoryRegistry` so that it has the timestamps, and then propagate that information through to the frontend (likely to the explorer).

**Comment by decentralion** [comment]
The first step here is to update the `repositoryRegistry` to have the new information.
Currently it stores a readonly array of `RepoIds`, as specified here: 
https://github.com/sourcecred/sourcecred/blob/master/src/core/repoIdRegistry.js#L15
`export type RepoIdRegistry = $ReadOnlyArray<RepoId>;`

We'll want to update this so that instead of just storing the `RepoId`, it stores an entry that contains a timestamp, e.g. 
```
export type RepoIdEntry = {|
  +repoId: RepoId,
  +timestampMs: number
|}
```
Changing the type signature will break the existing code in a few places; `yarn flow` will reveal all the fixups that are needed.

Since this will change a data format that is used in production and serialized to disk, we'll also need to update the [compat](https://github.com/sourcecred/sourcecred/blob/master/src/util/compat.js) information. We can do this by changing the compat version in [this line](https://github.com/sourcecred/sourcecred/blob/master/src/core/repoIdRegistry.js#L13) from 0.1.0 to 0.2.0, so that anyone with the old data format but the new source code will need to regenerate their sourcecred_directory.

**Comment by wchargin** [comment]
> We'll want to update this so that instead of just storing the RepoId, it stores an entry that contains a timestamp, e.g.

I have a patch for the first part of this that I'm happy to submit

**Comment by wchargin** [comment]
> We should be able to use `Date.now()` to get the time in ms (@wchargin
> maybe you will have a thought on a better API for this?)

Yeah, `Date.now()` is a fine implementation, as is `+new Date()` (which
is what I tend to use).

It would be best for this to be dependency-injected to facilitate
testing, as done in the Mirror module: the `update` function takes a
function `now: () => Date`, which can be implemented with a stub
function in tests or the simple `() => new Date()` in production. This
way, you don’t need to deal with mocking out globals, and the code that
you’re testing is the code that actually runs in production. Mocking in
a light e2e/integration test is fine.

**Comment by decentralion** [comment]
Once someone has added the last update information to the repositoryRegistry, it will become available on the frontend. The Explorer App has responsibility for loading the repo registry from the backend, so it can also display this information. Its [State module] actually loads the data; that could be a reasonable spot to load the repoRegistry. Or potentially the explorer could load the registry itself, it's not clear that it needs to live in State. Here's the explorer's [render method].

We no longer have code that explicitly loads the repoRegistry in the frontend, but [here][registry] is how it worked before it was removed in #988.

Once this is added to the frontend, it should of course be tested, [like the rest of the explorer][tests].

[registry]: https://github.com/sourcecred/sourcecred/blob/738853cd02ab3fd029196be800fb1ed7f92e9710/src/explorer/RepositorySelect.js#L78

[State module]: https://github.com/sourcecred/sourcecred/blob/bbe773bb67ef6132554335c643d676fb9d4fef9d/src/explorer/state.js#L27
[render method]: https://github.com/sourcecred/sourcecred/blob/bbe773bb67ef6132554335c643d676fb9d4fef9d/src/explorer/App.js#L86
[tests]: https://github.com/sourcecred/sourcecred/blob/master/src/explorer/App.test.js
